src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,476 @@
+
+    this.divContainer = React.createRef();
+    this.svgContainer = React.createRef();
+    this.anchor = React.createRef();
+
+    this.xAxisScale;
+    this.yAxisScale;
+    this.xAxis;
+    this.yAxisL;
+    this.yAxisR;
+    this.gx;
+    this.gyr;
+    this.gyl;
+    this.streams;
+
+    this.hoveredKey;
+    this.hoveredValue;
+    this.mouseIsOverStream = false;
+
+    this.zoomFunction = zoomFunction.bind(this);
+  }
+
+  initTheGraphBasicsElements(width, height, margin, svgViewport) {
+    const zoom = d3
+      .zoom()
+      .scaleExtent([1, 32])
+      .translateExtent([[0, 0], [width, height]])
+      .extent([[0, 0], [width, height]])
+      .on("zoom", this.zoomFunction);
+
+    const innerSpace = svgViewport
+      .append("g")
+      .attr("width", width)
+      .attr("class", "inner_space")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+    innerSpace
+      .append("defs")
+      .append("clipPath")
+      .attr("id", "mask")
+      .append("rect")
+      .attr("width", width)
+      .attr("height", height);
+
+    const graphZone = innerSpace
+      .append("g")
+      .attr("width", width)
+      .attr("height", height)
+      .attr("clip-path", "url(#mask)")
+      .call(zoom);
+
+    graphZone
+      .append("rect")
+      .attr("width", "100%")
+      .attr("height", "100%")
+      .attr("opacity", 0);
+
+    return [innerSpace, graphZone];
+  }
+
+  stackDatas(nameList, valuesArray) {
+    let stackMethod = d3
+      .stack()
+      .keys(nameList)
+      .order(d3.stackOrderNone)
+      .offset(d3.stackOffsetWiggle);
+
+    let stackedData;
+    if (valuesArray.length > 0) {
+      stackedData = stackMethod(valuesArray);
+    }
+    return stackedData;
+  }
+
+  createAndSetAxis(
+    minDate,
+    maxDate,
+    minValue,
+    maxValue,
+    width,
+    height,
+    innerSpace
+  ) {
+    minDate = new Date(String(minDate));
+    maxDate = new Date(String(maxDate));
+
+    this.xAxisScale = d3
+      .scaleTime()
+      .domain([minDate, maxDate])
+      .range([0, width]);
+
+    this.xAxis = d3
+      .axisBottom(this.xAxisScale)
+      .tickFormat(d3.timeFormat("%Y"))
+      .ticks(d3.timeYear, 1);
+
+    this.gx = innerSpace
+      .append("g")
+      .attr("class", "x axis")
+      .attr("transform", "translate(0," + height + ")")
+      .call(this.xAxis);
+
+    this.yAxisScale = d3
+      .scaleLinear()
+      .domain([minValue, maxValue])
+      .range([height, 0]);
+
+    this.yAxisL = d3.axisLeft(this.yAxisScale);
+    this.yAxisR = d3.axisRight(this.yAxisScale);
+
+    this.gyr = innerSpace
+      .append("g")
+      .attr("class", "y axis")
+      .attr("transform", "translate(" + width + ", 0)")
+      .call(this.yAxisR);
+
+    this.gyl = innerSpace
+      .append("g")
+      .attr("class", "y axis")
+      .call(this.yAxisL);
+  }
+
+  createAndSetStreams(layersNumber, graphZone, stackedData, nameList) {
+    let colorNameList = []; // e.g.: [Thematique,#FFFFFF]
```

Comment:
You are forced to use a comment to explain what is inside of that value.
But an object would do this for you!

```js
const colorNameList = [];
colorNameList.push({ name, color });
```

In general, we prefer to use objects instead of lists.

Reply:
yes, python habits here, that's fixed.